src/HPWH.cc

```diff
@@ -2151,6 +2264,7 @@ int HPWH::HPWHinit_file(string configFile){
 
 	//clear out old stuff if you're re-initializing
 	if (tankTemps_C != NULL) delete[] tankTemps_C;
+	if (tempTemps != NULL) delete[] tempTemps;
 	if (setOfSources != NULL) delete[] setOfSources;
 

```

Comment:
if (p != NULL) delete[] p -- the NULL test is not needed, delete[] does nothing when p == NULL.  However, depending on the situation, it is often desirable to add p = NULL after the delete[] to be absolutely sure p will not be delete[]ed again.  Or use stdlib smart pointers.

Also, there seem to be several places where these arrays are allocated or deleted.  Should there be functions such as AllocArrays() and DeleteArrays() that are called from the several places?  That approach makes changes easier/safer -- no need to review the entire file, just change the functions.

Reply:
There's many places these arrays are allocated and deleted, and I'm not exactly sure why for each case. I'll remove NULL tests, but going through the entire file can probably wait.